engine/src/flutter/shell/platform/darwin/common/framework/Source/Logger.swift

```diff
@@ -0,0 +1,132 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Darwin
+import Foundation
+
+/// The level of logging severity.
+///
+/// These levels are used by `Logger` to determine if a message should be output.
+/// They are ordered by increasing severity.
+@objc(FlutterLogLevel) public enum LogLevel: Int {
+  /// Informational messages that are helpful for tracing application flow.
+  case info = 0
+
+  /// Messages indicating a potential issue or an unexpected situation that isn't critical.
+  case warning
+
+  /// Messages indicating a runtime error from which the application can potentially recover.
+  case error
+
+  /// Messages that, while not an error, highlight significant progress or state changes in the
+  /// application, and must be logged.
+  case important
+
+  /// Messages indicating a critical condition. Causes the application to immediately terminate.
+  case fatal
+}
+
+/// A singleton logger for outputting runtime messages.
+///
+/// This logger allows for messages to be logged at different severity levels. Its output can be
+/// filtered by setting the `logLevel` property to the minimum log level to be logged.
+///
+/// **Usage:**
+/// ```swift
+/// Logger.logInfo("Application has started.")
+/// Logger.logLevel = .warning  // Only show warnings and above
+/// Logger.logError("Failed to load asset: \(assetKey)")
+/// ```
+@objc(FlutterLogger) public final class Logger: NSObject {
+  private static var shared = Logger()
+  private let outputWriter: OutputWriter
+  public let logLevel = LogLevel.info
+
+  public init(outputWriter: OutputWriter, logLevel: LogLevel) {
+    self.outputWriter = outputWriter
```

Comment:
Looks like this is mistakenly checked-in? (`logLevel` parameter is not used). I will pause the review and look again when it's ready.

Reply:
Not checked in but good catch -- missed that line before pushing.